files/SZWESAMP/ZWEGENER

```diff
@@ -0,0 +1,89 @@
+//ZWEGENER JOB                                                                  
+//*                                                                             
+//* This job is responsible for generating other jobs required                  
+//* to configure Zowe.                                                          
+//*                                                                             
+//* The method of validating your configuration is using
+//* JSON Schema <https://json-schema.org>. Zowe provides
+//* the ConfigMgr to assist in this. This job will invoke
+//* the ConfigMgr to validate your current configuration
+//* before generating any jobs. If there are any values
+//* that are incorrect, you will be notified. You should
+//* fix the value and then run this job again. You can run
+//* this job as many times as you need.
+//*
+//* Configmgr documentation:
+//* https://docs.zowe.org/stable/user-guide/configmgr-using
+//*
+//* Note: Any string with braces has an associated yaml value
+//*       in one of the yaml definitions for Zowe.
+//*       You must find the value and substitute it.
+//*
+//*       {key} -> value
+//*
+//GENER    EXEC PGM=IKJEFT1B
+//ISPPROF  DD   DSN=,DISP=(NEW,DELETE),UNIT=,
+//   DCB=(RECFM=FB,LRECL=80,BLKSIZE=3120,DSORG=PO),
+//   SPACE=(3120,(20,5,10))
+//*
+//* Replace {zowe.setup.dataset.prefix} with the
+//* Value as seen in zowe.yaml
+//*
+//SYSPROC  DD   DSN={zowe.setup.dataset.prefix}.SZWEEXEC,DISP=SHR
+//*
+//* Replace {zowe.setup.dataset.prefix} with the
+//* Value as seen in zowe.yaml
+//*
+//STEPLIB  DD   DSN={zowe.setup.dataset.prefix}.SZWELOAD,DISP=SHR
+//ISPPLIB  DD   DSN=ISP.SISPPENU,DISP=SHR
+//ISPMLIB  DD   DSN=ISP.SISPMENU,DISP=SHR
+//ISPTLIB  DD   DSN=ISP.SISPTENU,DISP=SHR
+//ISPSLIB  DD   DSN=ISP.SISPSENU,DISP=SHR
+//*
+//* The order must be as follows.
+//*
+//* zowe-yaml-schema.json
+//* server-common.json
+//*
+//* Replace {zowe.runtimeDirectory} with where your Zowe run time
+//* directory is, as seen in zowe.yaml
+//*
+//MYSCHEMA DD   *,DLM=$$
+FILE {zowe.runtimeDirectory}/schemas/zowe-yaml-schema.json
+FILE {zowe.runtimeDirectory}/schemas/server-common.json
+$$
+//*
+//* The DD below must include one or more FILE or PARMLIB
+//* Entries. The lower entries have their values
+//* Overridden by the higher entries.
+//* PARMLIB member must be named "ZWEYAML"
```

Comment:
* Prepare complete yaml into `ZOWE.PARMLIB(MYPARM)`
* Run `zwe init generate -c "PARMLIB(ZOWE.PARMLIB(MYPARM))"`
* JCL is generated and executed, somewhere in output is:
```
ZWEL0318E - failed to read '//'ZOWE.PARMLIB(ZWEYAML)'' - EDC5067I An attempt was made to open a nonexistent file for read.                                                                                                 
```
`ZWEYAML` is mandatory, but you can use anything else without warning or error. 

From my point of view, this is confusing:
* `zwe init generate` knows, the JCL will fail...
* ...or even worse, JCL will not fail, if there is already `ZWEYAML` member in the dataset, which will be used instead of `MYPARM`

Reply:
Oh, I misunderstood and now understand the point.

Reply:
Just happened while testing new change, I used randomly some old parmlib member:
`zwe init generate -c 'PARMLIB(MARTIN(TEST))'`
```
Error: Validation of PARMLIB(MARTIN):FILE(migrate2Jcl/files/defaults.yaml) against schema /migrate2Jcl/schemas/zowe-yaml-schema.json:/migrate2Jcl/schemas/server-common.json found invalid JSON Schema data
Validity Exceptions(s) with object at
  Validity Exceptions(s) with object at /zowe
    Validity Exceptions(s) with object at /zowe/setup
      Validity Exceptions(s) with object at /zowe/setup/dataset
        Validity Exceptions(s) with object at /zowe/setup/dataset/parmlibMembers
          Validity Exceptions(s) with string at /zowe/setup/dataset/parmlibMembers/zis
            string pattern match fail s='HELLO', pat='^ZWESIP[A-Z0-9\$\#\@]{2}$', at /zowe/setup/dataset/parmlibMembers/zis
```

This parmlib member was create before I have added schema validation for `zowe.setup.dataset.parmlibMembers.zis`.

It does not matter, it can happen to customer with another field.

So I have corrected the error, run it again and got:
```
Temporary directory '/tmp/.zweenv-8550' created.
Zowe will remove it on success, but if zwe exits with a non-zero code manual cleanup would be needed.
ERROR: ZWEL0318E Configuration stored in PARMLIB must use member name ZWEYAML when using generate action.
```

Technically it is working, but not a good user experience. `getZoweConfig` is called to soon.